src/pybind/mgr/devicehealth/module.py

```diff
@@ -275,7 +275,7 @@ def serve(self):
                     next_scrape = now
                 else:
                     # align to scrape interval
-                    scrape_frequency = int(self.scrape_frequency) or 86400
+                    scrape_frequency = int(self.scrape_frequency)
```

Comment:
I think `int(self.scrape_frequency)` is just a red herring. what 
```py
scrape_frequency = int(self.scrape_frequency) or 86400
```
tries to do is

- as you noticed, to cast `self.scrape_frequency` to int. but I doubt if this is necessary. as `PyModule::load_options()` is able to support the typed option and it casts the options to the specified type when loading them.
- probably you missed, to avoid the divide-by-zero error, as latter we have `seconds -= seconds % scrape_frequency`. in case user specifies "0" as the frequency of scraping.

Reply:
Thank you for letting me know. I'll close this PR.